libyul/Object.h

```diff
@@ -116,6 +117,21 @@ struct Object: public ObjectNode
 	/// The path must not lead to a @a Data object (will throw in that case).
 	std::vector<size_t> pathToSubObject(YulString _qualifiedName) const;
 
+	/// Searches for a subobject at @param _qualifiedName within the current object.
+	/// @returns a pointer to the subobject or a nullptr if it was not found.
```

Comment:
Please clarify what happens when the path leads to a `Data` object.

Also, it's not entirely true that it returns null if not found. The function has a check against that but it will trigger an assert in `Object::at()` before it even gets to that check.